tools/build_api.py

```diff
@@ -284,13 +300,24 @@ def get_mbed_official_release(version):
 
     return mbed_official_release
 
-ARM_COMPILERS = ("ARM", "ARMC6", "uARM")
 def target_supports_toolchain(target, toolchain_name):
-    if toolchain_name in ARM_COMPILERS:
-        return any(tc in target.supported_toolchains for tc in ARM_COMPILERS)
+    if int(target.build_tools_metadata["version"]) > 0:
+        if toolchain_name in target.supported_toolchains:
+            return True
+        else:
+            if(toolchain_name == "ARM"):
+                #we cant find ARM, see if one ARMC5, ARMC6 or uARM listed
+                return any(tc in target.supported_toolchains for tc in ("ARMC5","ARMC6","uARM"))
+            if(toolchain_name == "ARMC6"):
+                #we did not find ARMC6, but check for ARM is listed
+                return any(tc in target.supported_toolchains for tc in ("ARM",))
```

Comment:
Nit: simplification and readability

```suggestion
                return "ARM" in target.supported_toolchains
```

Reply:
Fixed